packages/cli/src/commands/inspect/index.ts

```diff
@@ -92,20 +99,46 @@ export default async function inspect(client: Client) {
   );
 
   const until = Date.now() + timeout;
-  const wait = argv['--wait'];
+  const wait = parsedArguments.flags['--wait'] ?? false;
+  const withLogs = parsedArguments.flags['--logs'];
 
   // resolve the deployment, since we might have been given an alias
   let deployment = await getDeployment(client, contextName, deploymentIdOrHost);
 
+  let abortController: AbortController | undefined;
+  if (withLogs) {
+    if (wait) {
+      abortController = displayBuildLogs(client, deployment, true);
+    } else {
+      await displayBuildLogs(client, deployment, false);
+      return exitCode(deployment.readyState, print);
+    }
+  }
+
   while (Date.now() < until) {
-    if (!wait || !isDeploying(deployment.readyState)) {
+    if (!wait) {
```

Comment:
Nit: maybe wrap the `while` with `if (wait)` or check `wait` in the `while` condition instead of breaking inside the while